eclipse-collections/src/main/java/org/eclipse/collections/impl/AbstractRichIterable.java

```diff
@@ -763,4 +763,22 @@ public boolean containsAll(Collection<?> collection)
         this.forEach(new GroupByUniqueKeyProcedure<>(target, function));
         return target;
     }
+
+    @Override
+    public T getOnly()
+    {
+        int size = this.size();
+
+        if (size == 0)
+        {
+            throw new IllegalStateException("Iterable is empty.");
+        }
+
+        if (size > 1)
+        {
+            throw new IllegalStateException("Iterable has multiple values.");
+        }
+
+        return this.getFirst();
```

Comment:
An exception like this would require fewer lines and would provide more information - the size.

```
        if (this.size() == 1)
        {
            return this.getFirst();
        }
        throw new IllegalStateException("Size must be 1 but was " + this.size());
```

Another advantage is that we wouldn't be using the word iterable to refer to specific subclasses.